### ISSUE TITLE
Git submodules using `git` urls can break `cargo install`

### ISSUE THREAD

**Issue by Hoverbear** [containers/youki#1212]
Currently if a user attempts to install `youki` via Cargo, they may meet an error if they do not have git keys configured fully:

```bash
❯ cargo install --git https://github.com/containers/youki
    Updating git repository `https://github.com/containers/youki`
    Updating git submodule `git@github.com:opencontainers/runtime-tools.git`
error: failed to update submodule `integration_tests/oci-runtime-tests`

Caused by:
  failed to fetch submodule `integration_tests/oci-runtime-tests` from git@github.com:opencontainers/runtime-tools.git

Caused by:
  failed to authenticate when downloading repository

  * attempted ssh-agent authentication, but no usernames succeeded: `git`

  if the git CLI succeeds then `net.git-fetch-with-cli` may help here
  https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli
```

It may be desirable to use a different Git submodule remote URL protocol.

While attempting to build `youki` in a sandbox without configured git keys I notably saw this error which I think might be related?: 

```rust
youki>    Compiling futures-executor v0.3.24
youki>    Compiling futures v0.3.24
youki> error: failed to run custom build command for `youki v0.0.3 (/build/source/crates/youki)`
youki> Caused by:
youki>   process didn't exit successfully: `/build/source/target/release/build/youki-e2aad9c6a21f0e82/build-script-build` (exit status: 1)
youki>   --- stderr
youki>   Error: could not find repository from '/build/source/crates/youki'; class=Repository (6); code=NotFound (-3)
youki> warning: build failed, waiting for other jobs to finish...
error: builder for '/nix/store/4769q3a9n0d3ay1qmbha0q5brvg9jv6q-youki-0.0.3.drv' failed with exit code 101;
       last 10 log lines:
       >    Compiling youki v0.0.3 (/build/source/crates/youki)
       >    Compiling futures-executor v0.3.24
       >    Compiling futures v0.3.24
       > error: failed to run custom build command for `youki v0.0.3 (/build/source/crates/youki)`
       >
       > Caused by:
       >   process didn't exit successfully: `/build/source/target/release/build/youki-e2aad9c6a21f0e82/build-script-build` (exit status: 1)
       >   --- stderr
       >   Error: could not find repository from '/build/source/crates/youki'; class=Repository (6); code=NotFound (-3)
       > warning: build failed, waiting for other jobs to finish...
       For full logs, run 'nix log /nix/store/4769q3a9n0d3ay1qmbha0q5brvg9jv6q-youki-0.0.3.drv'.
```

**Comment by YJDoc2** [comment]
Hey @Hoverbear the first issue will probably be solved by using https instead of ssh for git submodules, however, that particular submodule isn't even necessary for using youki, so I'm looking into some ways we can make it optional . 

As for the  second issue, can you provide the setup to reproduce it, and possibly the complete log (the log says `last 10 log lines`)?
Thanks!

**Comment by Hoverbear** [comment]
Yeah! Let me get a reproduction branch this weekend. :) Sorry about the wait.

**Comment by Hoverbear** [comment]
You can reproduce this with `nix run github:hoverbear/nixpkgs/init-youki#youki --print-build-logs --extra-experimental-features "nix-command flakes"` if you have `nix` installed.

<details>
  <summary>Full log</summary>
  
  ```
❯ nix build .#youki --print-build-logs
warning: Git tree '/home/ana/git/nixos/nixpkgs' is dirty
youki> unpacking sources
youki> unpacking source archive /nix/store/dmbfhyjbyilz7aainn3c0flq4kp3y57n-source
youki> source root is source
youki> Executing cargoSetupPostUnpackHook
youki> unpacking source archive /nix/store/fw2902zjk7l3caamsb5vq547mwm3xdcl-youki-0.0.3-vendor.tar.gz
youki> Finished cargoSetupPostUnpackHook
youki> patching sources
youki> Executing cargoSetupPostPatchHook
youki> Validating consistency between /build/source//Cargo.lock and /build/youki-0.0.3-vendor.tar.gz/Cargo.lock
youki> Finished cargoSetupPostPatchHook
youki> configuring
youki> building
youki> Executing cargoBuildHook
youki> ++ env CC_x86_64-unknown-linux-gnu=/nix/store/k6yka69n28j47wjhm14h41r2kbv262y6-gcc-wrapper-11.3.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/store/k6yka69n28j47wjhm14h41r2kbv262y6-gcc-wrapper-11.3.0/bin/c++ CC_x86_64-unknown-linux-gnu=/nix/store/k6yka69n28j47wjhm14h41r2kbv262y6-gcc-wrapper-11.3.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/store/k6yka69n28j47wjhm14h41r2kbv262y6-gcc-wrapper-11.3.0/bin/c++ cargo build -j 32 --target x86_64-unknown-linux-gnu --frozen --release
youki> warning: /build/source/crates/libcgroups/Cargo.toml: version requirement `0.8.3+v0.7.0` for dependency `libbpf-sys` includes semver metadata which will be ignored, removing the metadata is recommended to avoid confusion
youki>    Compiling libc v0.2.133
youki>    Compiling proc-macro2 v1.0.45
youki>    Compiling unicode-ident v1.0.4
youki>    Compiling quote v1.0.21
youki>    Compiling syn v1.0.101
youki>    Compiling autocfg v1.1.0
youki>    Compiling cfg-if v1.0.0
youki>    Compiling version_check v0.9.4
youki>    Compiling pkg-config v0.3.25
youki>    Compiling log v0.4.17
youki>    Compiling anyhow v1.0.65
youki>    Compiling memchr v2.5.0
youki>    Compiling serde_derive v1.0.145
youki>    Compiling thiserror v1.0.37
youki>    Compiling serde v1.0.145
youki>    Compiling once_cell v1.15.0
youki>    Compiling fnv v1.0.7
youki>    Compiling strsim v0.10.0
youki>    Compiling protobuf v2.27.1
youki>    Compiling bitflags v1.3.2
youki>    Compiling ident_case v1.0.1
youki>    Compiling crossbeam-utils v0.8.11
youki>    Compiling pin-utils v0.1.0
youki>    Compiling futures-core v0.3.24
youki>    Compiling ppv-lite86 v0.2.16
youki>    Compiling regex-syntax v0.6.27
youki>    Compiling serde_json v1.0.85
youki>    Compiling io-lifetimes v0.7.3
youki>    Compiling tinyvec_macros v0.1.0
youki>    Compiling futures-channel v0.3.24
youki>    Compiling crc32fast v1.3.2
youki>    Compiling futures-task v0.3.24
youki>    Compiling ryu v1.0.11
youki>    Compiling heck v0.4.0
youki>    Compiling adler v1.0.2
youki>    Compiling itoa v1.0.3
youki>    Compiling futures-sink v0.3.24
youki>    Compiling rustix v0.35.11
youki>    Compiling futures-util v0.3.24
youki>    Compiling rustversion v1.0.9
youki>    Compiling libseccomp-sys v0.2.1
youki>    Compiling futures-io v0.3.24
youki>    Compiling num_threads v0.1.6
youki>    Compiling hashbrown v0.12.3
youki>    Compiling iana-time-zone v0.1.50
youki>    Compiling semver v1.0.13
youki>    Compiling unicode-bidi v0.3.8
youki>    Compiling linux-raw-sys v0.0.46
youki>    Compiling procfs v0.14.1
youki>    Compiling percent-encoding v2.2.0
youki>    Compiling os_str_bytes v6.3.0
youki>    Compiling pin-project-lite v0.2.9
youki>    Compiling termcolor v1.1.3
youki>    Compiling lazy_static v1.4.0
youki>    Compiling scopeguard v1.1.0
youki>    Compiling textwrap v0.15.1
youki>    Compiling byteorder v1.4.3
youki>    Compiling hex v0.4.3
youki>    Compiling sysinfo v0.26.4
youki>    Compiling crossbeam-queue v0.3.6
youki>    Compiling fixedbitset v0.4.2
youki>    Compiling path-clean v0.1.0
youki>    Compiling no-std-net v0.6.0
youki>    Compiling fastrand v1.8.0
youki>    Compiling either v1.8.0
youki>    Compiling uuid v1.1.2
youki>    Compiling unicode-width v0.1.10
youki>    Compiling tinyvec v1.6.0
youki>    Compiling miniz_oxide v0.5.4
youki>    Compiling proc-macro-error-attr v1.0.4
youki>    Compiling proc-macro-error v1.0.4
youki>    Compiling form_urlencoded v1.1.0
youki>    Compiling memoffset v0.6.5
youki>    Compiling num-traits v0.2.15
youki>    Compiling num-integer v0.1.45
youki>    Compiling slab v0.4.7
youki>    Compiling indexmap v1.9.1
youki>    Compiling crossbeam-epoch v0.9.10
youki>    Compiling clap_lex v0.2.4
youki>    Compiling pnet_base v0.31.0
youki>    Compiling tabwriter v1.2.1
youki>    Compiling libdbus-sys v0.2.2
youki>    Compiling libseccomp v0.2.3
youki>    Compiling flate2 v1.0.24
youki>    Compiling crossbeam-channel v0.5.6
youki>    Compiling rustc_version v0.4.0
youki>    Compiling aho-corasick v0.7.19
youki>    Compiling crossbeam-deque v0.8.2
youki>    Compiling unicode-normalization v0.1.22
youki>    Compiling crossbeam v0.8.2
youki>    Compiling test_framework v0.0.1 (/build/source/tests/rust-integration-tests/test_framework)
youki>    Compiling idna v0.3.0
youki>    Compiling regex v1.6.0
youki>    Compiling url v2.3.1
youki>    Compiling jobserver v0.1.25
youki>    Compiling time v0.3.14
youki>    Compiling getrandom v0.2.7
youki>    Compiling time v0.1.44
youki>    Compiling nix v0.25.0
youki>    Compiling atty v0.2.14
youki>    Compiling num_cpus v1.13.1
youki>    Compiling dbus v0.9.6
youki>    Compiling mio v0.8.4
youki>    Compiling filetime v0.2.17
youki>    Compiling pnet_sys v0.31.0
youki>    Compiling xattr v0.2.3
youki>    Compiling which v4.3.0
youki>    Compiling pentacle v1.0.0
youki>    Compiling rand_core v0.6.4
youki>    Compiling cc v1.0.73
youki>    Compiling tar v0.4.38
youki>    Compiling rand_chacha v0.3.1
youki>    Compiling env_logger v0.8.4
youki>    Compiling rand v0.8.5
youki>    Compiling libz-sys v1.1.8
youki>    Compiling libgit2-sys v0.13.4+1.4.2
youki>    Compiling vergen v7.4.2
youki>    Compiling quickcheck v1.0.3
youki>    Compiling protobuf-codegen v2.27.1
youki>    Compiling darling_core v0.14.1
youki>    Compiling protobuf-codegen-pure v2.27.1
youki>    Compiling thiserror-impl v1.0.37
youki>    Compiling getset v0.1.2
youki>    Compiling futures-macro v0.3.24
youki>    Compiling clap_derive v3.2.18
youki>    Compiling serde_repr v0.1.9
youki>    Compiling enum-iterator-derive v1.1.0
youki>    Compiling clap_derive v4.0.1
youki>    Compiling prctl v1.0.0
youki>    Compiling rust-criu v0.2.0
youki>    Compiling darling_macro v0.14.1
youki>    Compiling git2 v0.14.4
youki>    Compiling enum-iterator v1.1.3
youki>    Compiling caps v0.5.4
youki>    Compiling clap v3.2.22
youki>    Compiling darling v0.14.1
youki>    Compiling derive_builder_core v0.11.2
youki>    Compiling derive_builder_macro v0.11.2
youki>    Compiling futures-executor v0.3.24
youki>    Compiling futures v0.3.24
youki>    Compiling derive_builder v0.11.2
youki>    Compiling clap_complete v3.2.5
youki>    Compiling liboci-cli v0.0.3 (/build/source/crates/liboci-cli)
youki>    Compiling youki v0.0.3 (/build/source/crates/youki)
youki> error: failed to run custom build command for `youki v0.0.3 (/build/source/crates/youki)`
youki> Caused by:
youki>   process didn't exit successfully: `/build/source/target/release/build/youki-2ef4e91e4709e861/build-script-build` (exit status: 1)
youki>   --- stderr
youki>   Error: could not find repository from '/build/source/crates/youki'; class=Repository (6); code=NotFound (-3)
youki> warning: build failed, waiting for other jobs to finish...
error: builder for '/nix/store/kmwn3my12avklj2im13lw0q1l8vx2zvr-youki-0.0.3.drv' failed with exit code 101;
       last 10 log lines:
       >    Compiling clap_complete v3.2.5
       >    Compiling liboci-cli v0.0.3 (/build/source/crates/liboci-cli)
       >    Compiling youki v0.0.3 (/build/source/crates/youki)
       > error: failed to run custom build command for `youki v0.0.3 (/build/source/crates/youki)`
       >
       > Caused by:
       >   process didn't exit successfully: `/build/source/target/release/build/youki-2ef4e91e4709e861/build-script-build` (exit status: 1)
       >   --- stderr
       >   Error: could not find repository from '/build/source/crates/youki'; class=Repository (6); code=NotFound (-3)
       > warning: build failed, waiting for other jobs to finish...
       For full logs, run 'nix log /nix/store/kmwn3my12avklj2im13lw0q1l8vx2zvr-youki-0.0.3.drv'.
  ```
</details>

**Comment by YJDoc2** [comment]
Hey @Hoverbear Thanks a lot for the complete logs! I am not much familiar with nix, so trying to work it out, however I have two questions :
- the nix run command you gave points to a repo at your github acc, I checked the repo manually but couldn't find anything related to youki. I haven't run the nix run command, and I don't know how it works, so might be my bad, but by any chance have you not pushed any changed you have made for youki to it? The logs also mention `git tree is dirty`
- Another thing I noticed is that the error is related to git, and after some checking, I feel that it is originating originally from git2, via vergen which we use to populate some compile time vars such as commit sha etc in the binary. Now again, I do not have much idea about how nix works, but from the log line `unpacking source archive /nix/store/fw2902zjk7l3caamsb5vq547mwm3xdcl-youki-0.0.3-vendor.tar.gz` , I think the build is extracting files from a tar gzip and using them? In that case would a .git be setup along with it, because if not then that might explain the `could not find repository` error.

It'd be great if you can help me for above, in the meantime, I'll also try digging a bit more to see what might be issue.
Thanks ! :)

**Comment by Hoverbear** [comment]
> the nix run command you gave points to a repo at your github acc, I checked the repo manually but couldn't find anything related to youki

That'd be because it's a branch! https://github.com/hoverbear/nixpkgs/tree/init-youki

That command would build this: https://github.com/Hoverbear/nixpkgs/blob/init-youki/pkgs/applications/virtualization/youki/default.nix

I think you're right about the missing `.git`.

Earlier in the build (I forced a clean cache) I noticed it downloaded: https://github.com/containers/youki/archive/c5ab61fcfdbd5d9b5e63009a9a17073332fca868.tar.gz

Which does not have a `.git.

I updated the code to do a proper git fetch, however Nix's Flake purity system means only things tracked in `git` get copied over, which does not include `.git`. So that's also a no go.

Is it possible to use a fallback? I see there is a `skip_if_error` option: https://docs.rs/vergen/latest/vergen/struct.Build.html#method.skip_if_error_mut

**Comment by YJDoc2** [comment]
Hey, for the testing I have made a branch which has a potential fix that will avoid issues when .git is not present. However, note that it will work only is you're are running `cargo build` ,  as the makefile needs git for somethings.
Can you check if the issue resolved by my branch : https://github.com/YJDoc2/youki/tree/vergen-fix
I'm not sure if this will get merged as it is / at all in the main, but if it works with this, I can try opening a PR.
Thanks :)

**Comment by Hoverbear** [comment]
That does fix that specific issue! I get a different error now, but it does fix this!

<details>
  <summary>Full log</summary>
  
  ```
❯ nix run .#youki --print-build-logs --extra-experimental-features "nix-command flakes" --refresh
warning: Git tree '/home/ana/git/nixos/nixpkgs' is dirty
youki> unpacking sources
youki> unpacking source archive /nix/store/lx94g1v424qdx8k2s80m3672ln8fddzy-source
youki> source root is source
youki> Executing cargoSetupPostUnpackHook
youki> unpacking source archive /nix/store/wybw9a0sqp9dgi90fwxhcj73vdn49l28-youki-0.0.3-vendor.tar.gz
youki> Finished cargoSetupPostUnpackHook
youki> patching sources
youki> Executing cargoSetupPostPatchHook
youki> Validating consistency between /build/source//Cargo.lock and /build/youki-0.0.3-vendor.tar.gz/Cargo.lock
youki> Finished cargoSetupPostPatchHook
youki> configuring
youki> building
youki> Executing cargoBuildHook
youki> ++ env CC_x86_64-unknown-linux-gnu=/nix/store/k6yka69n28j47wjhm14h41r2kbv262y6-gcc-wrapper-11.3.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/store/k6yka69n28j47wjhm14h41r2kbv262y6-gcc-wrapper-11.3.0/bin/c++ CC_x86_64-unknown-linux-gnu=/nix/store/k6yka69n28j47wjhm14h41r2kbv262y6-gcc-wrapper-11.3.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/store/k6yka69n28j47wjhm14h41r2kbv262y6-gcc-wrapper-11.3.0/bin/c++ cargo build -j 32 --target x86_64-unknown-linux-gnu --frozen --release
youki> warning: /build/source/crates/libcgroups/Cargo.toml: version requirement `0.8.3+v0.7.0` for dependency `libbpf-sys` includes semver metadata which will be ignored, removing the metadata is recommended to avoid confusion
youki>    Compiling libc v0.2.133
youki>    Compiling proc-macro2 v1.0.46
youki>    Compiling quote v1.0.21
youki>    Compiling unicode-ident v1.0.4
youki>    Compiling syn v1.0.101
youki>    Compiling autocfg v1.1.0
youki>    Compiling cfg-if v1.0.0
youki>    Compiling pkg-config v0.3.25
youki>    Compiling version_check v0.9.4
youki>    Compiling log v0.4.17
youki>    Compiling anyhow v1.0.65
youki>    Compiling serde_derive v1.0.145
youki>    Compiling memchr v2.5.0
youki>    Compiling serde v1.0.145
youki>    Compiling thiserror v1.0.37
youki>    Compiling bitflags v1.3.2
youki>    Compiling fnv v1.0.7
youki>    Compiling strsim v0.10.0
youki>    Compiling ident_case v1.0.1
youki>    Compiling protobuf v2.27.1
youki>    Compiling crossbeam-utils v0.8.12
youki>    Compiling pin-utils v0.1.0
youki>    Compiling futures-core v0.3.24
youki>    Compiling ppv-lite86 v0.2.16
youki>    Compiling regex-syntax v0.6.27
youki>    Compiling tinyvec_macros v0.1.0
youki>    Compiling futures-channel v0.3.24
youki>    Compiling crc32fast v1.3.2
youki>    Compiling io-lifetimes v0.7.3
youki>    Compiling futures-task v0.3.24
youki>    Compiling serde_json v1.0.85
youki>    Compiling futures-util v0.3.24
youki>    Compiling adler v1.0.2
youki>    Compiling heck v0.4.0
youki>    Compiling rustix v0.35.11
youki>    Compiling futures-sink v0.3.24
youki>    Compiling itoa v1.0.3
youki>    Compiling ryu v1.0.11
youki>    Compiling rustversion v1.0.9
youki>    Compiling procfs v0.14.1
youki>    Compiling iana-time-zone v0.1.50
youki>    Compiling num_threads v0.1.6
youki>    Compiling linux-raw-sys v0.0.46
youki>    Compiling os_str_bytes v6.3.0
youki>    Compiling libseccomp-sys v0.2.1
youki>    Compiling percent-encoding v2.2.0
youki>    Compiling unicode-bidi v0.3.8
youki>    Compiling semver v1.0.13
youki>    Compiling hashbrown v0.12.3
youki>    Compiling pin-project-lite v0.2.9
youki>    Compiling futures-io v0.3.24
youki>    Compiling termcolor v1.1.3
youki>    Compiling hex v0.4.3
youki>    Compiling once_cell v1.15.0
youki>    Compiling textwrap v0.15.1
youki>    Compiling scopeguard v1.1.0
youki>    Compiling lazy_static v1.4.0
youki>    Compiling byteorder v1.4.3
youki>    Compiling crossbeam-queue v0.3.6
youki>    Compiling sysinfo v0.26.4
youki>    Compiling fixedbitset v0.4.2
youki>    Compiling fastrand v1.8.0
youki>    Compiling no-std-net v0.6.0
youki>    Compiling path-clean v0.1.0
youki>    Compiling either v1.8.0
youki>    Compiling uuid v1.1.2
youki>    Compiling unicode-width v0.1.10
youki>    Compiling tinyvec v1.6.0
youki>    Compiling miniz_oxide v0.5.4
youki>    Compiling form_urlencoded v1.1.0
youki>    Compiling clap_lex v0.2.4
youki>    Compiling proc-macro-error-attr v1.0.4
youki>    Compiling proc-macro-error v1.0.4
youki>    Compiling memoffset v0.6.5
youki>    Compiling num-traits v0.2.15
youki>    Compiling slab v0.4.7
youki>    Compiling num-integer v0.1.45
youki>    Compiling indexmap v1.9.1
youki>    Compiling crossbeam-epoch v0.9.11
youki>    Compiling pnet_base v0.31.0
youki>    Compiling tabwriter v1.2.1
youki>    Compiling libdbus-sys v0.2.2
youki>    Compiling libseccomp v0.3.0
youki>    Compiling flate2 v1.0.24
youki>    Compiling crossbeam-channel v0.5.6
youki>    Compiling rustc_version v0.4.0
youki>    Compiling aho-corasick v0.7.19
youki>    Compiling crossbeam-deque v0.8.2
youki>    Compiling unicode-normalization v0.1.22
youki>    Compiling crossbeam v0.8.2
youki>    Compiling test_framework v0.0.1 (/build/source/tests/rust-integration-tests/test_framework)
youki>    Compiling regex v1.6.0
youki>    Compiling idna v0.3.0
youki>    Compiling getrandom v0.2.7
youki>    Compiling nix v0.25.0
youki>    Compiling time v0.1.44
youki>    Compiling num_cpus v1.13.1
youki>    Compiling atty v0.2.14
youki>    Compiling dbus v0.9.6
youki>    Compiling mio v0.8.4
youki>    Compiling filetime v0.2.17
youki>    Compiling pnet_sys v0.31.0
youki>    Compiling xattr v0.2.3
youki>    Compiling which v4.3.0
youki>    Compiling pentacle v1.0.0
youki>    Compiling url v2.3.1
youki>    Compiling jobserver v0.1.25
youki>    Compiling time v0.3.14
youki>    Compiling rand_core v0.6.4
youki>    Compiling tar v0.4.38
youki>    Compiling cc v1.0.73
youki>    Compiling rand_chacha v0.3.1
youki>    Compiling env_logger v0.8.4
youki>    Compiling rand v0.8.5
youki>    Compiling libz-sys v1.1.8
youki>    Compiling libgit2-sys v0.13.4+1.4.2
youki>    Compiling vergen v7.4.2
youki>    Compiling quickcheck v1.0.3
youki>    Compiling protobuf-codegen v2.27.1
youki>    Compiling protobuf-codegen-pure v2.27.1
youki>    Compiling darling_core v0.14.1
youki>    Compiling prctl v1.0.0
youki>    Compiling thiserror-impl v1.0.37
youki>    Compiling getset v0.1.2
youki>    Compiling futures-macro v0.3.24
youki>    Compiling clap_derive v3.2.18
youki>    Compiling serde_repr v0.1.9
youki>    Compiling enum-iterator-derive v1.1.0
youki>    Compiling clap_derive v4.0.8
youki>    Compiling rust-criu v0.2.0
youki>    Compiling darling_macro v0.14.1
youki>    Compiling git2 v0.14.4
youki>    Compiling enum-iterator v1.1.3
youki>    Compiling caps v0.5.4
youki>    Compiling clap v3.2.22
youki>    Compiling darling v0.14.1
youki>    Compiling derive_builder_core v0.11.2
youki>    Compiling derive_builder_macro v0.11.2
youki>    Compiling liboci-cli v0.0.3 (/build/source/crates/liboci-cli)
youki>    Compiling clap_complete v3.2.5
youki>    Compiling derive_builder v0.11.2
youki>    Compiling futures-executor v0.3.24
youki>    Compiling futures v0.3.24
youki>    Compiling youki v0.0.3 (/build/source/crates/youki)
youki> warning: An Error occurred during processing of Git. VERGEN_GIT_* may be incomplete.
youki>    Compiling chrono v0.4.22
youki>    Compiling syscalls v0.6.6
youki>    Compiling ipnetwork v0.19.0
youki>    Compiling pnet_datalink v0.31.0
youki>    Compiling oci-spec v0.5.8
youki>    Compiling libcgroups v0.0.3 (/build/source/crates/libcgroups)
youki>    Compiling runtimetest v0.0.1 (/build/source/tests/rust-integration-tests/runtimetest)
youki>    Compiling libcontainer v0.0.3 (/build/source/crates/libcontainer)
youki> error[E0599]: no method named `get_notify_fd` found for struct `ScmpFilterContext` in the current scope
youki>    --> crates/libcontainer/src/seccomp/mod.rs:205:17
youki>     |
youki> 205 |             ctx.get_notify_fd()
youki>     |                 ^^^^^^^^^^^^^ method not found in `ScmpFilterContext`
youki> For more information about this error, try `rustc --explain E0599`.
youki> error: could not compile `libcontainer` due to previous error
youki> warning: build failed, waiting for other jobs to finish...
error: builder for '/nix/store/cfnyijzq2li1lspigcl4b1jbnmcgrvq1-youki-0.0.3.drv' failed with exit code 101;
       last 10 log lines:
       >    Compiling libcontainer v0.0.3 (/build/source/crates/libcontainer)
       > error[E0599]: no method named `get_notify_fd` found for struct `ScmpFilterContext` in the current scope
       >    --> crates/libcontainer/src/seccomp/mod.rs:205:17
       >     |
       > 205 |             ctx.get_notify_fd()
       >     |                 ^^^^^^^^^^^^^ method not found in `ScmpFilterContext`
       >
       > For more information about this error, try `rustc --explain E0599`.
       > error: could not compile `libcontainer` due to previous error
       > warning: build failed, waiting for other jobs to finish...
       For full logs, run 'nix log /nix/store/cfnyijzq2li1lspigcl4b1jbnmcgrvq1-youki-0.0.3.drv'.
  ```
</details>

**Comment by YJDoc2** [comment]
Hey @Hoverbear , that is interesting. For this I'll have to run the nix build myself and do some debugging, as I do not have any guesses why this might be failing. I'll try to do it this weekend, and get back here. Thanks a lot for reporting this and giving full log!

**Comment by YJDoc2** [comment]
Hey, Sure! I'll keep this issue open till weekend and then close but please let know if any other issues, or if it works! :)

**Comment by YJDoc2** [comment]
Hey, unfortunately I didn't get to build with nix yet, and it might take some time. Meanwhile, the git url issue has to be solved separately with changes in what make targets are, I will open a separate issue for that.

As the vargen issue was solved with the merge of #1251 in main, closing this issue for now. If you have any issue, feel free to re-open or open a new issue.

Thanks for trying to build youki with nix, I got a lot to learn while working on this!
